tests/utils/pod.go

```diff
@@ -115,7 +115,7 @@ func findPodByCompFuncOnce(clientSet *kubernetes.Clientset, namespace, prefix, l
 	for _, pod := range podList.Items {
 		if compFunc(pod.Name, prefix) {
 			if result == nil {
-				result = &pod
+				result = pod.DeepCopy()
```

Comment:
why ref is not good enough here?

Reply:
It has to do with how go uses the index/value in range loops. If you just use the ref, then next loop the ref will be the new value. But we want to store the old value. You can see it in the output if there is another pod with the same prefix. Both names printed will be the same. So instead I copy the value and store that.

Reply:
Just hit this on the 1.38 release branch, this means `result` will hold last pod that we iterated through (can be any pod in the namespace), do we want to backport this small fix?

Reply:
This particular fix is for debug printing the right pod info. Did you run into a failure and it printed the wrong info? We can backport this but doesn't seem that useful.

Reply:
This can give you the wrong pod when using `FindPodByPrefix()` which is bad